### PR TITLE
IDToken now parses using nimbus-jose

### PR DESCRIPTION
**Pull request by iambmelt**
See title

### CODE REVIEW

**Comment by iambmelt** [comment]
Upstream changes in MSAL needed to make this work in unit tests

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java [hunk 1]
```diff
@@ -219,27 +217,27 @@ public String getRawIDToken() {
...
-        } else {
-            throw new IllegalArgumentException("Invalid ID token format.");
+            for (final Map.Entry<String, Object> entry : claimsMap.entrySet()) {
+                claimsMapStr.put(entry.getKey(), entry.getValue().toString());
+            }
+
+            // Return our result
+            return claimsMapStr;
+        } catch (ParseException e) {
+            throw new ServiceException("Failed to parse JWT", ErrorStrings.INVALID_JWT, null);
```
Comment (addev-ashish) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java]: Can we move this catch block to line 227 and pass the ParseException to ServiceException

**Comment by iambmelt** [comment]
@addev-ashish I wasn't planning on resolving the `ClientInfo` case as part of this PR. If you'd like to see that change here lmk, otherwise I'll stick with the simple IdToken parsing case

**Comment by iambmelt** [comment]
@addev-ashish Feedback addressed

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java [hunk 2]
```diff
@@ -219,27 +217,29 @@ public String getRawIDToken() {
...
-
+    public static Map<String, String> parseJWT(final String idToken) throws ServiceException {
+        final JWTClaimsSet claimsSet;
         try {
-            final String decodedBody = new String(data, Charset.forName(StringExtensions.ENCODING_UTF8));
-            return JsonExtensions.extractJsonObjectIntoMap(decodedBody);
-        } catch (final JSONException e) {
-            throw new ServiceException("", ErrorStrings.INVALID_JWT, e);
+            // Create a SignedJWT from the input token String
+            final SignedJWT signedJWT = SignedJWT.parse(idToken);
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java]: @kreedula This should follow this guidance to support v1's unsigned jwts: https://connect2id.com/products/nimbus-jose-jwt/examples/combined-jwt-parsing

**Pull request merged**